tmcprototype/DishLeafNode/DishLeafNode/DishLeafNode.py

```diff
@@ -279,14 +304,124 @@ def convert_radec_to_azel(self, data):
+    #         if (self.el <= self.ele_min_lim or self.el > self.ele_max_lim) or self.event_track_time.isSet():
+    #             print("In ele lim thread")
+    #             self.event_el.set()
+    #             self._read_activity_message = CONST.ERR_ELE_LIM
+    #             print(CONST.ERR_ELE_LIM)
+    #             break
+
+    def tracking_time_thread(self):
+        """This thread allows the dish to track the source for a specified Duration.
+
+        :return: None.
+
+        """
+        start_track_time = time.time()
+        end_track_time = start_track_time + self.TrackDuration * 60.0
+        while 1:
+            if end_track_time <= time.time():
+                self.event_track_time.set()
+                self._read_activity_message = CONST.ERR_TIME_LIM
+                print(CONST.ERR_TIME_LIM)
+                break
+            elif self.el_limit == True:
+                break
+
+    def track_thread(self, argin):
+        """This thread invokes Track command on DishMaster at the rate of 20 Hz.
+
+        :param argin: DevVarStringArray
+
+        For Track thread, argin to be provided is the Ra and Dec values in the following format:
+        radec|2:31:50.91|89:15:51.4 Where first value is tag that is radec, second value is Ra in Hr:Min:Sec,
+        and third value is Dec in Deg:Min:Sec.
+
+        It takes system's current time in UTC as timestamp and converts RaDec to AzEl using
+        convert_radec_to_azel method of class DishLeafNode.
+
+        :return: None.
+
+        """
+        radec_value = argin.replace('|', ',')
+        # RaDec as input argument
+
+        #Timestamp value if given as input argument
+        #timestamp_value = argin[1].replace('|', ' ')
+
+        try:
+            while self.event_track_time.is_set() is False:
+                # timestamp_value = Current system time in UTC
+                timestamp_value = str(datetime.datetime.utcnow())
+                katpoint_arg = []
+                katpoint_arg.insert(0, radec_value)
+                katpoint_arg.insert(1, timestamp_value)
+                # Conversion of RaDec to AzEl
+                self.convert_radec_to_azel(katpoint_arg)
+                if self.RaDec_AzEl_Conversion is True:
+                    if self.el >= 17.5 and self.el <= 90:
+                        if self.az < 0:
+                            self.az = 360 - abs(self.az)
+
+                        roundoff_az_el = [round(self.az, 12), round(self.el, 12)]
+                        spectrum = [0]
+                        spectrum.extend((roundoff_az_el))
+                        # assign calculated AzEl to desiredPointing attribute of Dishmaster
+                        self._dish_proxy.desiredPointing = spectrum
+                        # Invoke Track command of Dish Master
+                        print("Desired Pointing:", self._dish_proxy.desiredPointing)
+                        self._dish_proxy.command_inout_asynch("Track", "0", self.commandCallback)
```

Comment:
Instead of string "Track", define it in CONST.py file and use it. For eg: refer CONST.CMD_TRACK = "Track"

Reply:
Done. Commit ID :2e978037512d9ba47e7b5d8f8bf016ce6633d4dd